### PR TITLE
DDF-2645 Update Tika transformer and metacard attributes for Mp4 files

### PR DESCRIPTION
**Pull request by mcalcote**
#### What does this PR do?
Follows on the previous Tika transformer improvements to include Mp4 specific attributes.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@clockard @brianfelix @gordocanchola 

#### Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).
[Data](https://github.com/orgs/codice/teams/data) @brendan-hofmann 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@coyotesqrl
@figliold

#### How should this be tested? (List steps with links to updated documentation)
Confirm build/itests pass.

#### What are the relevant tickets?
[DDF-2645](https://codice.atlassian.net/browse/DDF-2645)

#### Checklist:
- [ ] Documentation Updated
- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/Mp4Attributes.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.impl.AttributeDescriptorImpl;
+import ddf.catalog.data.impl.BasicTypes;
+
+/**
+ * This class provides attributes that reflect Mp4 attributes expected to be relevant to all metacard types
+ */
+public class Mp4Attributes implements MetacardType {
```
Comment (emmberk) [inline catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/Mp4Attributes.java]: I'm not sure about this, but should there be a corresponding interface like `Media` for `MediaAttributes`?

File: catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/Mp4Attributes.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    private static final Set<AttributeDescriptor> DESCRIPTORS;
+
+    private static final String NAME = "mp4";
+
+    private static final String EXT_PREFIX = "ext.mp4.";
+
+    public static final String AUDIO_SAMPLE_RATE = EXT_PREFIX + "audio-sample-rate";
```
Comment (brendan-hofmann) [inline catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/Mp4Attributes.java]: We'll need to work on formalizing our `ext.*` process for incorporation into the taxonomy proper after 2.10, @jlcsmith . 😄

File: catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/Mp4Attributes.java (hunk 1 above)
Comment (clockard) [inline catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/Mp4Attributes.java]: Generally we would call this class something like `Mp4MetacardType`

File: catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/Mp4Attributes.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    static {
+        Set<AttributeDescriptor> descriptors = new HashSet<>();
+        descriptors.add(new AttributeDescriptorImpl(AUDIO_SAMPLE_RATE,
+                true /* indexed */,
+                true /* stored */,
+                false /* tokenized */,
+                false /* multivalued */,
+                BasicTypes.STRING_TYPE));
```
Comment (clockard) [inline catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/Mp4Attributes.java]: Should this be a string or a number of some sort?

File: catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/Mp4Attributes.java (hunk 1 above)
Comment (emmberk) [inline catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/Mp4Attributes.java]: It looks like we do call classes that implement `MetacardType` `*Attributes` a lot of the time. Is there a convention for this?

File: catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/Mp4Attributes.java (hunk 3 above)
Comment (emmberk) [inline catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/Mp4Attributes.java]: Looks like it should be a number, right? https://tika.apache.org/1.8/api/org/apache/tika/metadata/XMPDM.html#AUDIO_SAMPLE_RATE

File: catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/Mp4Attributes.java (hunk 1 above)
Comment (clockard) [inline catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/Mp4Attributes.java]: Seems that is the case for the common taxonomy ones but it looks like most all of the other ones (especially in upstream projects) use the MetacardType naming convention. You can go either way here I guess.

Comment (mcalcote) [inline catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/Mp4Attributes.java]: Originally, I modeled this class after the attributes classes in the taxonomy, but based on the comments to this PR, I plan to extend the MetacardTypeImpl and rename the class to `Mp4MetacardType`.

File: catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/Mp4Attributes.java (hunk 2 above)
Comment (mcalcote) [inline catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/Mp4Attributes.java]: :+1:

File: catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/Mp4Attributes.java (hunk 1 above)
Comment (mcalcote) [inline catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/Mp4Attributes.java]: Updated to `*MetacardType`

File: catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/Mp4Attributes.java (hunk 3 above)
Comment (mcalcote) [inline catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/Mp4Attributes.java]: `Can be any value` doesn't instill confidence, but I double checked the spec through the link an it says the property is an integer type (p.25).

Updated

**Comment by mcalcote** [comment]
@coyotesqrl I did include some custodial changes in this PR. I can split the changes to the blueprint file into a separate PR/commits if that would be preferable.

**Comment by mcalcote** [comment]
The blueprint updates have been captured in PR #1603

**Comment by taz77** [comment]
Starting a build!

**Comment by taz77** [comment]
DDF hero build has passed all tests.

**Pull request merged**